CMakeLists.txt

```diff
@@ -113,128 +87,70 @@ if (NOT DEFINED VERONA_LLVM_LOCATION)
-    DEPENDS external
-    BUILD_ALWAYS true
-    USES_TERMINAL_BUILD true
-    USES_TERMINAL_CONFIGURE true
-    USES_TERMINAL_INSTALL true
-    USES_TERMINAL_TEST true
-    CMAKE_ARGS ${VERONA_EXTRA_CMAKE_ARGS})
-
-  enable_testing()
-
-  add_subdirectory(testsuite)
-
-  # Main target does not test runtime, build subdirectory for that
-  if (RT_TESTS)
-    add_subdirectory(src/rt)
-  endif()
-
-  include(ProcessorCount)
-  ProcessorCount(N)
-
-  # Adds a target check that runs the tests.
-  add_custom_target(check
-    COMMAND ${CMAKE_CTEST_COMMAND} -C $<CONFIG> -j ${N} --output-on-failure --timeout 400 --interactive-debug-mode 0
-    USES_TERMINAL
-  )
-
-  # If RT_TESTS are enabled, we need to make sure they are
-  # up-to-date before we run tests.
-  if (RT_TESTS)
-    add_dependencies(check rt_tests)
-  endif()
-
-  return ()
 endif()
 
-project(verona-lang CXX)
-
-message(STATUS "Build Type for Verona ${CMAKE_BUILD_TYPE}")
-set(CMAKE_CONFIGURATION_TYPES  Release Debug RelWithDebInfo)
-message(STATUS "Build types ${CMAKE_CONFIGURATION_TYPES}")
-
-if (VERONA_CI_BUILD)
-  # Specify policy to go into child projects.
-  set (CMAKE_POLICY_DEFAULT_CMP0077 NEW)
-  set (SNMALLOC_CI_BUILD ON)
+##########################################################
+#  Build Verona subproject
+##########################################################
+if (DEFINED VERONA_EXTRA_CMAKE_ARGS)
+  message (WARNING "Do not use VERONA_EXTRA_CMAKE_ARGS! Value ignored")
 endif ()
-
-# Copied from LLVM, handling release builds with assertions isn't trivial as
-# CMake defines -DNDEBUG automatically and we have to clear it.
-if( ENABLE_ASSERTS )
-  # MSVC doesn't like _DEBUG on release builds.
-  if( NOT MSVC )
-    message(STATUS "Enabling asserts")
-    add_definitions( -D_DEBUG )
-  endif()
-  # On non-Debug builds cmake automatically defines NDEBUG, so we
-  # explicitly undefine it:
-  if( NOT uppercase_CMAKE_BUILD_TYPE STREQUAL "DEBUG" )
-    message(STATUS "Removing NDEBUG from existing flags to enable asserts")
-    # NOTE: use `add_compile_options` rather than `add_definitions` since
-    # `add_definitions` does not support generator expressions.
-    add_compile_options($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-UNDEBUG>)
-
-    # Also remove /D NDEBUG to avoid MSVC warnings about conflicting defines.
-    foreach (flags_var_to_scrub
-        CMAKE_CXX_FLAGS_RELEASE
-        CMAKE_CXX_FLAGS_RELWITHDEBINFO
-        CMAKE_CXX_FLAGS_MINSIZEREL
-        CMAKE_C_FLAGS_RELEASE
-        CMAKE_C_FLAGS_RELWITHDEBINFO
-        CMAKE_C_FLAGS_MINSIZEREL)
-      string (REGEX REPLACE "(^| )[/-]D *NDEBUG($| )" " "
-        "${flags_var_to_scrub}" "${${flags_var_to_scrub}}")
-    endforeach()
+set (VERONA_EXTRA_CMAKE_ARGS)
+
+# Point Verona at the MLIR
+list (APPEND VERONA_EXTRA_CMAKE_ARGS
+  -DVERONA_LLVM_LOCATION=${MLIR_INSTALL}/install
+)
+
+# Define project level defaults
+list (APPEND VERONA_EXTRA_CMAKE_ARGS
+  -DCMAKE_CXX_STANDARD=17)
```

Comment:
Should this just be define from within `src/CMakeLists.txt`?

Reply:
Done